File: q0455/q0455.java
class Solution {
    public int findContentChildren(int[] g, int[] s) {
        Arrays.sort(g);
        Arrays.sort(s);
        int cookieIndex = 0;
        int childIndex = 0;
        while (cookieIndex < s.length && childIndex < g.length) {
            // try to use the smallest cookie to match the current kid with smallest greed
            if (s[cookieIndex] >= g[childIndex]) {
                childIndex++;
            }
            // update the cookieIndex
            // case 1: satisfied the kid and need a new cookie for the next kid
            // case 2: kid's greed factor larger than current cookie, need a bigger cookie
            cookieIndex++;
        }
        return childIndex;
    }
}

File: q0455/q0455.py
def findContentChildren(g: List[int], s: List[int]) -> int:
    s.sort()
    g.sort()
    childIndex = cookieIndex = 0
    while childIndex < len(g) and cookieIndex < len(s):
        if s[cookieIndex] >= g[childIndex]:
            childIndex += 1
        # update the cookieIndex 
        # case 1: satisfied the kid and need a new cookie for the next kid
        # case 2: kid's greed factor larger than current cookie, need a bigger cookie
        cookieIndex += 1
    return childIndex
